Kafka/src/main/java/io/deephaven/kafka/publish/GenericRecordKeyOrValueSerializer.java

```diff
@@ -0,0 +1,723 @@
+            }
+
+            for (int ii = 0; ii < objectContext.inputChunk.size(); ++ii) {
+                final Object raw = objectContext.inputChunk.get(ii);
+                avroChunk.get(ii).put(fieldName, raw);
+            }
+        }
+    }
+
+    private class TimestampFieldProcessor extends GenericRecordFieldProcessor {
+        public TimestampFieldProcessor(String fieldName) {
+            super(fieldName);
+        }
+
+        @Override
+        FieldContext makeContext(int size) {
+            return null;
+        }
+
+        @Override
+        public void processField(FieldContext fieldContext,
+                WritableObjectChunk<GenericRecord, Attributes.Values> avroChunk, OrderedKeys keys, boolean isRemoval) {
+            // do we really want nanos instead of micros; there are avro things for micros/millis?
+            final long nanos = DBDateTime.now().getNanos();
+            for (int ii = 0; ii < avroChunk.size(); ++ii) {
+                avroChunk.get(ii).put(fieldName, nanos);
+            }
+        }
+    }
+
+    protected final List<GenericRecordFieldProcessor> fieldProcessors = new ArrayList<>();
+
+    public GenericRecordKeyOrValueSerializer(final DynamicTable source,
+            final Schema schema,
+            final Map<String, String> columnsToOutputFields,
+            final Set<String> excludedColumns,
+            final boolean autoValueMapping,
+            final boolean ignoreMissingColumns,
+            final String timestampFieldName) {
+        this.source = source;
+        this.schema = schema;
+
+        final String[] columnNames = source.getDefinition().getColumnNamesArray();
+        final List<String> missingColumns = new ArrayList<>();
+
+        // Create all the auto-mapped columns
+        for (final String columnName : columnNames) {
+            if (excludedColumns.contains(columnName) || columnsToOutputFields.containsValue(columnName)) {
+                continue;
+            }
+            if (autoValueMapping) {
+                makeFieldProcessor(columnName, columnName);
+            } else {
+                if (!ignoreMissingColumns) {
+                    missingColumns.add(columnName);
+                }
+            }
+        }
+
+        if (!missingColumns.isEmpty()) {
+            final StringBuilder sb = new StringBuilder("Found columns without mappings " + missingColumns);
+            if (!excludedColumns.isEmpty()) {
+                sb.append(", unmapped=").append(excludedColumns);
+            }
+            if (!columnsToOutputFields.isEmpty()) {
+                sb.append(", mapped to fields=").append(columnsToOutputFields.keySet());
+            }
+            throw new KafkaPublisherException(sb.toString());
+        }
+
+        // Now create all the processors for specifically-named fields
+        columnsToOutputFields.forEach(this::makeFieldProcessor);
+
+        if (!StringUtils.isNullOrEmpty(timestampFieldName)) {
```

Comment:
I don't think is obvious to me how this feature works.   We seem to be adding support for an additional "timestamp me this at the time of generation".  Which can come on either key, value, or both (by nature of our support).
The utility of that versus the Kafka native timestamp field, again, not obvious to me.